toolchain/sem_ir/typed_insts.h

```diff
@@ -793,6 +793,24 @@ struct ImplDecl {
   DeclInstBlockId decl_block_id;
 };
 
+// The type of an associated function within an `impl`. This is the type of
+// `(SelfType as Interface).AssociatedFunction`.
```

Comment:
I'm mildly concerned about confusion because we also have `impl fn`, which also matches this name. Had you considered something like "FunctionTypeInImpl" or (maybe incorrect) "AssociatedFunctionType"?

Reply:
Renamed to `FunctionTypeWithSelfType` for now.